backends/core/models.py

```diff
@@ -59,6 +61,13 @@ class Pipeline(BaseModel):
 
   def __init__(self, name=None):
     self.name = name
+    cached_values = {
+      "failed_jobs": 0, 
```

Comment:
Avoid using string literals for keys, it's more error prone. You can for example put a constant in the class like this:

```python
class Pipeline(BaseModel):
  ...
  CACHE_KEY_FAILED_JOBS = 'failed_jobs'
  CACHE_KEY_REMAINING_JOBS = 'remaining_jobs'
  CACHE_KEY_LIST_OF_TASKS_ENQUEUED = '...'
  ...
```

And then reference them as `self.CACHE_KEY_FAILED_JOBS` in your class methods.

Reply:
done